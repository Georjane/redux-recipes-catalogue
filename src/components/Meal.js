import PropTypes from 'prop-types';

function Meal(props) {
  const { meal } = props;
  const { strMeal, strMealThumb } = meal;

  return (
    <div className="meal">
      <img src={strMealThumb} alt="Logo" />
      <h2>{strMeal}</h2>
    </div>
  );
}

Meal.propTypes = {
  meal: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default Meal;
