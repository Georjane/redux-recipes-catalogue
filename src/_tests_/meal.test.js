import React from 'react';
import renderer from 'react-test-renderer';
import Meal from '../components/Meal';

it('renders meal page', () => {
  const strMeal = 'Chicken soup';
  const strMealThumb = 'https://www.themealdb.com/images/media/meals/sbx7n71587673021.jpg';
  const tree = renderer.create(<Meal meal={strMeal, strMealThumb } />);
  expect(tree).toMatchSnapshot();
});
