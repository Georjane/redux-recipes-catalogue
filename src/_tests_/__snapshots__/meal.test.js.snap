// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders meal page 1`] = `
<div
  className="meal"
>
  <img
    alt="Logo"
  />
  <h2 />
</div>
`;
